include/boost/geometry/arithmetic/arithmetic.hpp

```diff
@@ -51,17 +52,24 @@ struct value_operation
         : m_value(value)
     {}
 
-    template <typename P, int I>
-    inline void apply(P& point) const
+    template <typename PointDst, int I>
+    inline void apply(PointDst& dest_point) const
     {
```

Comment:
My primary goal was was to fix the issue so I've taken the names that were already used in this code. So `C` is a type of a value passed as the second argument to the `XXX_value()` functions. But sure, I'll fix the names.